tests/returntypes/test_adjoint_diff_new.py

```diff
@@ -0,0 +1,365 @@
+            qml.expval(qml.PauliZ(0))
+
+        tape.trainable_params = {1, 2, 3}
+
+        dM1 = dev.adjoint_jacobian_new(tape)
+
+        qml.execute([tape], dev, None)
+        dM2 = dev.adjoint_jacobian_new(tape, starting_state=dev._pre_rotated_state)
+
+        assert np.allclose(dM1, dM2, atol=tol, rtol=0)
+
+    def test_gradient_of_tape_with_hermitian(self, tol):
+        """Test that computing the gradient of a tape that obtains the
+        expectation value of a Hermitian operator works correctly."""
+        dev = qml.device("default.qubit", wires=3)
+
+        a, b, c = [0.5, 0.3, -0.7]
+
+        def ansatz(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.RX(c, wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+
+        mx = qml.matrix(qml.PauliX(0) @ qml.PauliY(2))
+        with qml.tape.QuantumTape() as tape:
+            ansatz(a, b, c)
+            qml.RX(a, wires=0)
+            qml.expval(qml.Hermitian(mx, wires=[0, 2]))
+
+        tape.trainable_params = {0, 1, 2}
+        res = dev.adjoint_jacobian_new(tape)
+
+        expected = [
+            np.cos(a) * np.sin(b) * np.sin(c),
+            np.cos(b) * np.sin(a) * np.sin(c),
+            np.cos(c) * np.sin(b) * np.sin(a),
+        ]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_multi_return(self, dev):
+        """Test that the gradients of multiple observables are correct"""
+
+        x = np.array([0.6, 0.8, 1.0])
+
+        ops = [
+            qml.Hadamard(wires=0),
+            qml.RX(0.543, wires=0),
+            qml.CNOT(wires=[0, 1]),
+            qml.Rot(x[0], x[1], x[2], wires=0),
+            qml.Rot(1.3, -2.3, 0.5, wires=[0]),
+            qml.RZ(-0.5, wires=0),
+            qml.RY(0.5, wires=1),
+            qml.CNOT(wires=[0, 1]),
+        ]
+
+        observables = [
+            qml.PauliX(0),
+            qml.PauliX(0) @ qml.PauliZ(1),
+            qml.Projector([0], wires=0),
+            qml.Hermitian([[0, 1], [1, 0]], wires=1),
+        ]
+
+        with qml.tape.QuantumTape() as tape:
+            for op in ops:
+                qml.apply(op)
+
+            for ob in observables:
+                qml.expval(ob)
+
+        tape.trainable_params = {1, 2, 3}
+
+        grad_D = dev.adjoint_jacobian_new(tape)
+
+        # check that the type and format of the adjoint jacobian is correct
+        assert isinstance(grad_D, tuple)
+        assert len(grad_D) == 4
+        assert isinstance(grad_D[0], np.ndarray)
+        assert grad_D[0].shape == (3,)
+
+        # check the results against individually executed tapes
+        for i, ob in enumerate(observables):
+            with qml.tape.QuantumTape() as indiv_tape:
+                for op in ops:
+                    qml.apply(op)
+
+                qml.expval(ob)
+
+            indiv_tape.trainable_params = {1, 2, 3}
+            assert np.allclose(grad_D[i], dev.adjoint_jacobian_new(indiv_tape))

```

Comment:
Minor: we may as well check the type and shape of every tuple element as extra checks.
```suggestion

        # check the results against individually executed tapes
        for i, ob in enumerate(observables):
            with qml.tape.QuantumTape() as indiv_tape:
                for op in ops:
                    qml.apply(op)

                qml.expval(ob)

            indiv_tape.trainable_params = {1, 2, 3}

            assert isinstance(grad_D[i], np.ndarray)
            assert grad_D[i].shape == (3,)
            assert np.allclose(grad_D[i], dev.adjoint_jacobian_new(indiv_tape))
```